workers/unity/Packages/com.improbable.gdk.mobile/Editor/LaunchMenu.cs

```diff
@@ -129,5 +130,214 @@ private static bool TryGetApkPath(string rootPath, out string apkPath)
+                    .WithArgs("-s", "devices")
+                    .AddOutputProcessing(message =>
+                    {
+                        if (deviceRegex.IsMatch(message))
+                        {
+                            availableDevices.Add(deviceRegex.Match(message).Value);
+                        }
+                    })
+                    .Run();
+
+                if (!TryGetXCTestRunPath(availableDevices.Count == 0, out xcTestRunPath))
+                {
+                    Debug.LogError("Unable to fund a xctestrun file for the correct architecture. Did you built it for the correct platform?");
+                    return;
+                }
+
+                if (!TryPrepareArguments(shouldConnectLocally, out var arguments))
+                {
+                    Debug.LogError("Failed to generate arguments");
+                    return;
+                }
+
+                // add SpatialOS arguments to iOS environment variables
+                var doc = new XmlDocument();
+                doc.Load(xcTestRunPath);
+                XmlNode spatialNode = null;
+                var rootNode = doc.DocumentElement.ChildNodes[0].ChildNodes[1];
+                for (var i = 0; i < rootNode.ChildNodes.Count; i++)
+                {
+                    var node = rootNode.ChildNodes[i];
+                    if (node.InnerText != "EnvironmentVariables")
+                    {
+                        continue;
+                    }
+
+                    var envValueNode = rootNode.ChildNodes[i + 1];
+                    for (var j = 0; j < envValueNode.ChildNodes.Count; j++)
+                    {
+                        var envNode = envValueNode.ChildNodes[j];
+                        if (envNode.InnerText != LaunchArguments.iOSEnvironmentKey)
+                        {
+                            continue;
+                        }
+
+                        spatialNode = node.ChildNodes[j + 1];
+                        break;
+                    }
+
+                    if (spatialNode != null)
+                    {
+                        spatialNode.InnerText = arguments;
+                    }
+                    else
+                    {
+                        var spatialKeyNode = doc.CreateNode("element", "key", string.Empty);
+                        spatialKeyNode.InnerText = LaunchArguments.iOSEnvironmentKey;
+                        var spatialValueNode = doc.CreateNode("element", "string", string.Empty);
+                        spatialValueNode.InnerText = arguments;
+                        envValueNode.AppendChild(spatialKeyNode);
+                        envValueNode.AppendChild(spatialValueNode);
+                    }
+
+                    break;
+                }
+
+                doc.Save(xcTestRunPath);
+
+                // fix document after updating it
+                var text = File.ReadAllText(xcTestRunPath);
+                text = text.Replace("[]>", ">");
```

Comment:
Why do we need to do this?

Reply:
because for some reason saving the doc adds a `[]` which makes it invalid. I haven't been able to figure out yet why it's doing that...

Reply:
Could you make a ticket for that and link it in a comment?